### PR TITLE
adding a docker image

### PR DESCRIPTION
**Pull request by joshuafernandes**
adding a docker image and circleci integration

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    steps:
+      - prepare
+      - attach_workspace:
+          at: ~/project
+      - setup_remote_docker
+      - run:
+          name: Build image
+          command: |
+            docker login --username "${DOCKER_USER_RW}" --password "${DOCKER_PASSWORD_RW}"
+            docker build -t consensys/istanbul-tools:$CIRCLE_SHA1 .
```
Comment (joshuafernandes) [inline .circleci/config.yml:50]: I'd like to use the version here but running the command `istanbul --version` gives me nothing useful, so using the hash

         jfernandes@falcon:~/workspace/istanbul-tools(docker-image)$ ./build/bin/istanbul --version
         istanbul version

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.15.3-buster
```
Comment (trung) [inline Dockerfile:1]: `istanbul-tools` depends on `quorum` so I would stick with 1.13 as we know it works. 
I have not tried 1.15

File: .circleci/config.yml (hunk 1 above)
Comment (trung) [reply .circleci/config.yml:50]: Currently there is no versioning for `istanbul-tools`. Maybe we can take this opportunity to establish one. IMO, using hash doesn't seem very user-friendly.

Comment (trung) [reply .circleci/config.yml:50]: Just revisit the versioning convention for `istanbul-tools`, indeed we do version it
```
› ./istanbul --version
istanbul version v1.0.3
```
It takes the Git Tag and inject into the code at build time.
We still need some work to modify it to use CalVer convention.

**Comment by joshuafernandes** [comment]
closing this for now, will come back to it when I have a chance to work on it

**Pull request closed without merging**